code/API_definitions/device-status.yaml

```diff
@@ -1,7 +1,72 @@
+      At least one identifier for the device must be provided from the following four options:
+      - IPv4 address
+      - IPv6 address
+      - Phone number (MSISDN)
+      - Network Access Identifier assigned by the mobile network operator for the device
+
+    * **Roaming** : Roaming status:
+      - `true` if device is connected to a network other than its home network
+      - `false` if connected to its home network
+
+    * **Country** : Visited country code and name:
+      - Mobile country code (MCC), as defined by ITU-T E.212 (e.g. `262`).
+      - ISO country code, as defined by ISO 3166 ALPHA-2 (e.g. `DE`). A list may be provided if multiple countries share the same MCC.
+
+
+    # API Functionality    
+
+    The API exposes following capabilities:
+
+    ## Device roaming status
+
+    The endpoint `POST /roaming` allows the API consumer to synchronously (i.e. immediately) get roaming status and country information (if the device is roaming). The response will include confirmation of the device identifier(s) used by the API implementation to determine the roaming status, but the response will never include additional identifiers that were not originally provided by the API consumer.
+
+    ## Device status event subscription
+
+    This endpoint allows the API consumer to create and delete event subscriptions for specific roaming device status events. The CAMARA subscription model is detailed in the [CAMARA API Design Guidelines](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#121-subscription).
+
+    Following `eventType` options are defined for this API: 
+      - `ROAMING_STATUS` - This event is triggered when the device roaming status changes from roaming ON (`true`) to roaming OFF (`false`) or vice-versa
+
+      - `ROAMING_ON` - This event is only triggered when the device changes from roaming OFF to roaming ON
+
+      - `ROAMING_OFF` - This event is only triggered when the device changes from roaming ON to roaming OFF
+
+      - `ROAMING_CHANGE_COUNTRY` - This event is triggered whenever the mobile country code (MCC) of the network to which the device is connected changes. Changing networks within a given country does not trigger this event.
+
+    ## Notification callback
+
+    This endpoint describes the event notification that will be received by the subscription listener (API consumer) side when a subscribed event has occurred. A detailed description of the event notification mechanism is provided in the [CAMARA API Design Guidelines](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#122-event-notification).
+
+    **WARNING**: This endpoint must be implemented on the listener (API consumer) side, as a `POST /notifications` request will be made by the device status resource server to the url provided by notification listener.
+
```

Comment:
Is /notifications mandatory?
Doesn't the webhook/notificationUrl define the complete url?

Reply:
Hi @akoshunyadi 

Good question. I had assumed that it was mandatory for the notification endpoint name to match the name defined in the API definition. So either the API consumer includes that in `notificationUrl`, or it gets added on automatically to whatever is specified.

However, looking at the example in the [CAMARA Design Guidelines](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#12-subscription-notification--event), I see that the API consumer specified `"notificationUrl": "https://application-server.com"`but the notification was sent to `https://application-server.com/v0/notifications`.

So now I'm asking "where did the `v0` come from?". Rules around constructing the full notification URL need to be clarified.

But, despite that, my assumption was that if the API consumer had instead specified `"notificationUrl": "https://application-server.com/notification"`, then that is exactly where the notification would be sent (rather than `"https://application-server.com/notification/notification"`).

Reply:
As per my understanding, when the API consumer registers its webhook/notification URL with the API provider, the provider should perform validation on the URL to ensure it meets the required format that should be properly documented at our end. If there are any issues, such as missing components or incorrect structure (missing “/notification”), the API provider should respond with an appropriate error message indicating what went wrong and how the API consumer can correct the URL.

Also it may create some confusion to have extra endpoint in our swagger file as  if client try to access “Try it out” feature then this extra endpoint will take him/her to  http://localhost:9091//device-status/v0/notifications) that can confuse him/her.

Reply:
In QoD we decided not to add or expect "/notifications", just use the provided callback url
notifications:
          "{$request.body#/webhook/notificationUrl}":

Reply:
Given that we have an explicit `/notifications` endpoint defined here, and given the resource-based subscription example used in the design guidelines, I think we have to assume that the notification endpoint must end with `/notifications`.

The question is what we should do if the `notificationUrl` provided by the API consumer does not end with `/notifications`? The example in the design guidelines suggests (more or less) that we should just add `/notifications` to the endpoint provided by the API consumer.

But I agree with @Sachinsiso that a better option might be to reject it and get the API consumer to try again to avoid the risk that the API consumer has misunderstood how to implement this endpoint. This interpretation would require a change in the design guidelines.

I'd suggest to wait until @bigludo7 returns before discussing further, and keep this PR open in the meantime.

Reply:
Hi,

will also align this with @fernandopradocabrillo, anycase I think it is good to wait until Ludovic returns. 

I consider the comment is valid, and probably this can generate misunderstanding. When sending notification from Telco Operator to client, the endpoint is settled by {$request.body#/webhook/notificationUrl}, so really not required any append like "/notifications". The notification is sent to the endpoint provided by the client.

Think it is good time to have this discussion given there is an issue where a "traversal" API is being discussed/defined to cover this flow: https://github.com/camaraproject/Commonalities/issues/8